docs/security/annotations.md

```diff
@@ -0,0 +1,83 @@
+# Security Annotations
+
+Two security annotations are provided as an alternative to the security checks methods provided by the _AuthorizationManager_ interface.
+
+* **@ResourceCheck**
+
+```
+import static org.uberfire.shared.authz.ProjectConstants.*;
+...
+
+@ResourceCheck(type=PROJECT, action=CREATE)
+public void newProject() {
+    newProjectPresenter.show( this );
+}
+```
+
+Every time the method _newProject_ is called the security check defined in the annotation is processed, causing the method body to be executed only if the permission is granted.
+
+
+* **@PermsissionCheck**
+
+Annotation checks are supported for plain permissions as well.
+
+```
+@PermissionCheck("myapp.reports.sales")
+public void showSalesReport() {
+    salesReportsPresenter.show();
+}
+```
+
+#### Callbacks
+
+Both annotations provide callback support:
+
+```
+@ResourceCheck(type=PROJECT, action=CREATE,
+                onGranted="onCreateGranted",
+                onDenied="onCreateDenied")
```

Comment:
idk if that is possible with annotation, but IMO it would be better to pass the Method reference in his?

Something like:
@ResourceCheck(type=PROJECT, action=CREATE,
 +                onGranted=Class.method,
 +                onDenied="onCreateDenied")

again ^ idk if that is possible.